src/main/java/uk/gov/hmcts/darts/notification/service/impl/NotificationServiceImpl.java

```diff
@@ -35,28 +33,46 @@
 import java.util.List;
 import java.util.Map;
 
-
-@RequiredArgsConstructor
 @Service
 @Slf4j
 public class NotificationServiceImpl implements NotificationService {
 
-    private final NotificationRepository notificationRepo;
-
-    private final CaseRepository caseRepository;
-    private final GovNotifyService govNotifyService;
-    private final TemplateIdHelper templateIdHelper;
-    private final GovNotifyRequestHelper govNotifyRequestHelper;
-    private final EmailValidator emailValidator = EmailValidator.getInstance();
     private static final List<NotificationStatus> STATUS_ELIGIBLE_TO_SEND = Arrays.asList(
         NotificationStatus.OPEN,
         NotificationStatus.PROCESSING
     );
 
-    @Value("${darts.notification.max_retry_attempts}")
-    private int maxRetry;
-
+    private final NotificationRepository notificationRepo;
+    private final CaseRepository caseRepository;
+    private final GovNotifyService govNotifyService;
+    private final TemplateIdHelper templateIdHelper;
+    private final GovNotifyRequestHelper govNotifyRequestHelper;
     private final LogApi logApi;
+    private final boolean notificationsEnabled;
+    private final boolean atsMode;
+    private final int maxRetry;
+
+    public NotificationServiceImpl(
+        NotificationRepository notificationRepo,
+        CaseRepository caseRepository,
+        GovNotifyService govNotifyService,
+        TemplateIdHelper templateIdHelper,
+        GovNotifyRequestHelper govNotifyRequestHelper,
+        LogApi logApi,
+        @Value("${darts.notification.enabled}") boolean notificationsEnabled,
+        @Value("${darts.automated-tasks-pod}") boolean atsMode,
```

Comment:
I'll approve this , i'm sure there is a good reason but not sure why we didnt setup the configuration to prevent the class loading in the first place.